### ISSUE TITLE
Other languages support

### ISSUE THREAD

**Issue by InfroLab** [Babelscape/rebel#23]
Hey, outstanding paper you got. I mentioned it in my Thesis :)

I was trying to train it on Russian language. I've generated the dataset with CROCODILE. Edited config files. Downloaded facebook/mbart-50-large-50. Added it to conf files as well. Additionally, I added src_lang and tgt_lang parameter that equal ru_RU in AutoTokenizer in train.py

I struggled with numerous errors in rebel_short.py, pl_modules.py, pl_data_modules.py that happend on my setup. I also had to update pytorch-lightning several version up - to 1.3.0 because the version in requirements.txt got a problem at startup. May be it relates to https://github.com/Babelscape/rebel/issues/22

So, for now I have several questions that you as creators:
1) Should there be any problem with replacing bart with mbart-50?
2) What would you recommend to check if I see that a printed log with all my relations and their TP FP FN precision recall and F1 contains only zeros and the loss is nan?
3) In generate_samples.py there is a line 

pl_module.logger.experiment.log({"Triplets": wandb_table})

when everything is done and the training started at 999 iteration an exception is thrown there which says SummaryWriter() has no attribute log. I've checked the docs of pl and it does not really tell where should I look at least.

I put it try except block, but I worry that this may lead to 4th question

4) At 50% of training it stops with a RuntimeError 'could not infer dtype of Table' )))0

I am not sure if stacktraces will help here, they are not really informative.

Would love to get any answer here. Thanks again for the paper.

**Comment by LittlePea13** [comment]
Hi there, glad you enjoyed our work.
I will try my best with your questions:

1. Indeed, it should be possible to make it work with mbart-50, important stuff will be to deal properly with the special tokens, etc.
2. The extraction of the triplets. Check first what the model is generating after the generate function is called, and decode it to text. It may be that the model is behaving properly but there's some issue when using mbart50 that messes with the extract_triplets method. So my suggestion is to look at the raw text generated by the model after some training.
3. This has to do with the callback to report some predictions during training which were useful to debug and see how the model is learning. So if you are not using wandb, don't add that callback and it should be fine.
4. Since it seems to be related to question 4, I would first try to remove or comment out the line: https://github.com/Babelscape/rebel/blob/2eb4a3445c5f64b3be77f13e0122fff8c6d7d3cc/src/train.py#L84

Hope this helps, best of luck :)

**Comment by A1exRey** [comment]
Hi! I am also trying to remake the REBEL model for the Russian language. But I used for training the standard run_summarization.py script from HF and replaced mbart with ru-t5. It seems that the result is quite good (https://huggingface.co/memyprokotow/rut5-REBEL-base)

**Comment by InfroLab** [comment]
@LittlePea13 thanks for your detailed answers. I am not really getting what do you mean by dealing properly with special tokens? Do you mean that I should remove all unnecessary tokens that are added to mbart50? 

I would write down here if I have any progress in my attempts to make everything work.

@A1exRey Am I understanding right that you just changed the link to the model from bart to rut5 and changed the config? are there any other details? maybe you changed something else to make it work?

**Comment by InfroLab** [comment]
So, it turned out that the model is generating a pure and undoubtful crap:)
The input is like

"ru_RU Bla bla bla bla </s><pad><pad>...<pad>

and the output 
"<s>ru_RU<pad><pad>...<pad>" <<< all padding here

while the label is normal
"ru_RU <triplet> ... <subj> ... <obj> ... </s><pad><pad>...<pad>"

I guess this might be what you were talking about. <s> in output and </s> in label.

Having such output as above will make extract_triplets output just an empty list, obviously.

**Comment by A1exRey** [comment]
@InfroLab 
For training I used [HF summarization](https://github.com/huggingface/transformers/blob/main/examples/pytorch/summarization/run_summarization.py)
And added few lines of code

```
## line 407
tokenizer = AutoTokenizer.from_pretrained(
  additional_special_tokens= ['<obj>', '<subj>', '<triplet>'],
  extra_ids = 0,
```
And for training script
```
    --text_column context \
    --summary_column triplets \
    --max_source_length 512 \
    --max_target_length 512 \
    --variables_from https://github.com/Babelscape/rebel/blob/main/conf/train/default_train.yaml
```
I also loaded the Russian dataset for REBEL, if you need it. https://huggingface.co/datasets/memyprokotow/rebel-dataset-rus/viewer/memyprokotow--rebel-dataset-rus/train

P.S
It is possible that this code (via HF summarization and t5) is not quite correct, and it would be great if the author @LittlePea13  could confirm or deny this.

**Comment by InfroLab** [comment]
@A1exRey thanks for quick answer

I see a thing that I may be looking for. Authors are forcing bos token here https://github.com/huggingface/transformers/blob/c6cea5a78ce190858582cfe8be32b4c30e2b2a79/examples/pytorch/summarization/run_summarization.py#L478

**Comment by LittlePea13** [comment]
I will now close the issue if there is no further doubts, feel free to reopen it or share your results :)
